src/arithmetic/algebraic_expression/algebraic_expression_optimization.c

```diff
@@ -419,8 +522,15 @@ void AlgebraicExpression_Optimize
 	AlgebraicExpression **exp
 ) {
 	assert(exp);
+
 	_AlgebraicExpression_PushDownTranspose(*exp);
 	_AlgebraicExpression_MulOverSum(exp);
 	_AlgebraicExpression_FlattenMultiplications(*exp);
+
+	// We need to fetch operand matrices now, as the PerformTranspose optimization will utilize them.
+	_AlgebraicExpression_FetchOperands(*exp, QueryCtx_GetGraphCtx(), QueryCtx_GetGraph());
```

Comment:
@swilly22 Fetching operands here is a questionable choice, and I welcome your thoughts! We do require the matrices before the next call - the optimization does not make sense otherwise.

The alternative location for this call would be in ConditionalTraverse and ExpandInto before they call this function, but that would require changing this to an exposed interface and does not really seem preferable to me.

We can also just put this at the start of this function.

Reply:
I don't mind leaving `_AlgebraicExpression_FetchOperands` here, but let's move it to the top of this function, as the first "optimisation" performed.